include/ews/ews.hpp

```diff
@@ -11422,26 +12632,49 @@ namespace ews
     {
     public:
         // Intentionally not explicit
-        property_path(const char* uri) : uri_(std::string(uri)) {}
+        property_path(const char* uri) : uri_(uri) {
+            class_name(uri_);
+        }
 
         //! Returns the \<FieldURI> element for this property.
         //!
         //! Identifies frequently referenced properties by URI
-        const std::string& field_uri() const EWS_NOEXCEPT { return uri_; }
 
-        std::string property_name() const
+        std::string to_xml() const
+        {
+            std::stringstream sstr;
+            sstr << "<t:FieldURI FieldURI=\"";
+            sstr << uri_ << "\"/>";
+            return sstr.str();
+        }
+
+        std::string to_xml(const std::string& value) const
+        {
+            std::stringstream sstr;
+            sstr << "<t:FieldURI FieldURI=\"";
+            sstr << uri_ << "\"/>";
+            sstr << "<t:" << class_name(uri_) << ">";
+            sstr << "<t:" << property_name(uri_) << ">";
+            sstr << value;
+            sstr << "</t:" << property_name(uri_) << ">";
+            sstr << "</t:" << class_name(uri_) << ">";
+            return sstr.str();
+        }
+
+    protected:
+        static std::string property_name(const std::string& uri)
         {
-            const auto n = uri_.rfind(':');
+            const auto n = uri.rfind(':');
             EWS_ASSERT(n != std::string::npos);
-            return uri_.substr(n + 1);
+            return uri.substr(n + 1);
         }
 
-        std::string class_name() const
+        static std::string class_name(const std::string& uri)
```

Comment:
Why did you make this function a static member-function? Why not leave it a const member-function? No need for this. Revert. Also see `property_name` above.